.github/workflows/dotnet-ci.yml

```diff
@@ -135,3 +135,59 @@ jobs:
           --api-key dummy 
           --source PullRequests 
           --skip-duplicate
+
+  publish-coverage:
+    if: github.actor != 'dependabot[bot]'
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download test results
+        uses: actions/download-artifact@v4
+        with:
+          pattern: test-results*
+
+      - name: Combine coverage reports
+        uses: danielpalme/ReportGenerator-GitHub-Action@5.3.11
+        with:
+          reports: "**/*.cobertura.xml"
+          targetdir: "${{ github.workspace }}"
+          reporttypes: "Cobertura"
+          verbosity: "Info"
+          title: "Code Coverage"
+          tag: "${{ github.run_number }}_${{ github.run_id }}"
+          customSettings: ""
+          toolpath: "reportgeneratortool"
+
+      - name: Upload combined coverage XML
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage
+          path: ${{ github.workspace }}/Cobertura.xml
+          retention-days: 7
+
+      - name: Publish code coverage report
+        uses: irongut/CodeCoverageSummary@v1.3.0
+        with:
+          filename: "Cobertura.xml"
+          badge: true
+          fail_below_min: false
+          format: markdown
+          hide_branch_rate: false
+          hide_complexity: false
+          indicators: true
+          output: both
+          thresholds: "10 30"
```

Comment:
Is this the the min coverage and will anything under this break the CI?
Is possible for any packages to have their own the min coverage instead of using this one for all packages?

Reply:
This should fail the build according to https://github.com/irongut/CodeCoverageSummary/tree/v1.3.0/?tab=readme-ov-file#thresholds although I probably need to learn more about how to best set them (I copied this from the CommuntiyToolkit/Datasync repo 😏).

To have each package provide their own min threshold (like the dotnet/aspire repo has) we'd need something more custom here and that feels like we'd have to manage a bit more code. I think that we can get away with a blanket rule of "your coverage must meet X" (and generally speaking, we're pretty highly covered).

Reply:
I've adjusted the thresholds and it looks like it is marking things as failed but not the build yet, I think it'll only fail if the overall threshold drops below the min value, not if one of them is below. I'll keep experimenting